stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -2581,11 +2581,18 @@ impl NakamotoChainState {
 
         let active_pox_contract = pox_constants.active_pox_contract(burn_header_height.into());
 
+        let pox_reward_cycle = Burnchain::static_block_height_to_reward_cycle(
+            burn_header_height as u64,
+            sortition_dbconn.get_burn_start_height().into(),
+            sortition_dbconn.get_pox_reward_cycle_length().into(),
+        ).expect("FATAL: Unrecoverable chainstate corruption: Epoch 2.1 code evaluated before first burn block height");
```

Comment:
this would fail if the `burn_header_height` is less than 666050, which is when Stacks 2.0 started (not 2.1).